### PR TITLE
[hlm-1419]: Stock

### PR DESCRIPTION
**Pull request by shivab-egov**

### CODE REVIEW

**Review by jayantp-egov — CHANGES_REQUESTED** [review]

File: health-services/stock/src/main/java/org/egov/stock/config/MainConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import javax.annotation.PostConstruct;
+import java.util.TimeZone;
+@Import({TracerConfiguration.class})
+@Configuration
+@ComponentScan(basePackages = {"org.egov"})
+public class MainConfiguration {
```
[the remark this answers is not in the document]
Comment (shivab-egov) [reply health-services/stock/src/main/java/org/egov/stock/config/MainConfiguration.java:30]: resolved

File: health-services/stock/src/main/java/org/egov/stock/service/StockService.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    private final StockConfiguration configuration;
+
+    private final StockEnrichmentService enrichmentService;
+
+    private final Predicate<Validator<StockBulkRequest, Stock>> isApplicableForCreate =
+            validator -> validator.getClass().equals(SproductVaraintIdValidator.class);
+
+    private final Predicate<Validator<StockBulkRequest, Stock>> isApplicableForUpdate =
+            validator -> validator.getClass().equals(SproductVaraintIdValidator.class)
```
[the remark this answers is not in the document]
Comment (shivab-egov) [reply health-services/stock/src/main/java/org/egov/stock/service/StockService.java]: resolved

File: health-services/stock/src/main/java/org/egov/stock/service/StockService.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                              Integer offset,
+                              String tenantId,
+                              Long lastChangedSince,
+                              Boolean includeDeleted) throws Exception  {
+        String idFieldName = getIdFieldName(stockSearchRequest.getStock());
+        if (isSearchByIdOnly(stockSearchRequest.getStock(), idFieldName)) {
+            List<String> ids = (List<String>) ReflectionUtils.invokeMethod(getIdMethod(Collections
+                            .singletonList(stockSearchRequest.getStock())),
+                    stockSearchRequest.getStock());
+            return stockRepository.findById(ids, includeDeleted).stream()
```
[the remark this answers is not in the document]
Comment (shivab-egov) [reply health-services/stock/src/main/java/org/egov/stock/service/StockService.java]: resolved

File: health-services/stock/src/main/java/org/egov/stock/web/models/Stock.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+@Validated
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2023-02-08T11:49:06.320+05:30")
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Stock {
+    @JsonProperty("id")
+    private String id = null;
```
[the remark this answers is not in the document]
Comment (shivab-egov) [reply health-services/stock/src/main/java/org/egov/stock/web/models/Stock.java:29]: resolved

**Pull request merged**